src/sparseml/optim/helpers.py

```diff
@@ -195,6 +195,68 @@ def parse_recipe_variables(
     return recipe_variables
 
 
+def _update_staged_container_value(var_key, var_value, container):
+    """
+    Extends the functionality of _update_container_value to handle staged containers.
+
+    :param var_key: The key we are attempting to find in the container.
+    :param var_value: The value which will overwrite the previous value
+        every time var_key is found in the container's keys.
+    :param container: A container generated from a YAML string of SparseML recipe
+    :return: (optionally mutated) container, as well as key_value
+        (True if var_key found in container's attributes, otherwise False)
+    """
+    key_found = False
+
+    for container_key, container_value in container.items():
+        if not isinstance(container_value, dict):
+            # checking contents of global variables
+            if var_key == container_key:
+                container[var_key] = var_value
+                key_found = True
+
+        else:
+            # checking contents of a stage
+            stage_container, stage_key_found = _update_container_value(
+                var_key, var_value, container_value
+            )
+            container[container_key] = stage_container
+            if stage_key_found:
+                key_found = True
```

Comment:
if `stage_key_found` is guaranteed to be a bool, could do `key_found = key_found or stage_key_found` or `key_found |= stage_key_found`. not safe if you think `stage_key_found` could change type somehow

Reply:
The problem which I am trying to solve is the following. Imagine a container, where there is:
- one global variable.
- three stages. 
The key we are looking for is in the **second stage**.

if we apply `key_found = key_found or stage_key_found` the following thing might happen:
0. initially `key_found = False`
1. after checking one global variable `key_found = False` (we did not find the key)
2. after checking the first stage `stage_key_found = False` (variable not in the stage)
3. after checking the second stage `stage_key_found = True` (yay, variable found)
4. after checking the third stage `stage_key_found = False` (because no variable in the third stage).

We end up with `False or False`, even though the variable was present in the second stage!